rollup.config.js

```diff
@@ -1,37 +1,54 @@
-import commonjs from 'rollup-plugin-commonjs';
-import nodeResolve from 'rollup-plugin-node-resolve';
-import nodeBuiltins from 'rollup-plugin-node-builtins';
-import babel from 'rollup-plugin-babel';
-import banner from 'rollup-plugin-banner';
-import genHeader from './lib/header';
+import commonjs from 'rollup-plugin-commonjs'
+import resolve from 'rollup-plugin-node-resolve'
+import babel from 'rollup-plugin-babel'
+import typescript from 'rollup-plugin-typescript2'
+import banner from 'rollup-plugin-banner'
+import genHeader from './lib/header'
 
-export default [{
-	input: 'index.js',
-	output: {
-		file: 'dist/esm.js',
-		format: 'esm',
-	},
-	plugins: [
-		commonjs(),
-		nodeBuiltins(),
-		nodeResolve(),
-		babel(),
-		banner(genHeader('data'))
-	],
-},
-{
-	input: 'index.js',
-	output: {
-		file: 'dist/umd.js',
-		format: 'umd',
-		exports: 'named',
-		name: 'vis',
-	},
-	plugins: [
-		commonjs(),
-		nodeBuiltins(),
-		nodeResolve(),
-		babel(),
-		banner(genHeader('data'))
-	],
-}]
+// TypeScript because Babel transpiles modules in isolation, therefore no type reexports.
+// CommonJS because Babel is not 100 % ESM.
+
+const babelConfingBase = {
+  extensions: ['.mts', '.ts', '.tsx', '.mjs', '.js', '.jsx'],
+  runtimeHelpers: true,
+}
+const resolveConfig = {
+  extensions: [...babelConfingBase.extensions, '.json', '.node'],
+}
+const bannerConfig = genHeader('data')
+const typescriptConfig = {
+  tsconfig: 'tsconfig.code.json',
+}
+
+export default [
+  {
+    input: 'src/index.ts',
+    output: {
+      file: 'dist/esm.js',
+      format: 'esm',
+    },
+    plugins: [
+      resolve(resolveConfig),
+      typescript(typescriptConfig),
+      commonjs(),
+      babel(babelConfingBase),
+      banner(bannerConfig),
+    ],
+  },
+  {
+    input: 'src/index.ts',
+    output: {
+      file: 'dist/umd.js',
+      format: 'umd',
+      exports: 'named',
+      name: 'visData',
```

Comment:
I think we should export `vis` instead of `visData` to be compatible with earlier versions.

Reply:
I'm not sure about this, wouldn't there be conflicts between packages then? Like one package overwriting the other before it can be used? Anyway people using the old API won't import this package or will they? I assume that the exports from this package will be reexported under `vis` in vis-charts, vis-network and so on, keeping backwards compatibility.